### PR TITLE
docs: remove redundant section explaining authentication example

### PR DESCRIPTION
**Pull request by emonddr**
The authentication code example in
packages/authentication/docs/authentication-system.md can be
removed now that
https://loopback.io/doc/en/lb4/Loopback-component-authentication.html
exists and @loopback/authentication/README.MD has been updated.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/authentication/docs/authentication-system.md [hunk 1]
```diff
@@ -163,197 +163,3 @@ And the abstractions for:
...
-      }
-
-      this.reject(context, error);
-      return;
-    }
-  }
-}
-```
-
-Then custom sequence must be bound to the application, and the authentication
```
Comment (nabdelgadir) [inline packages/authentication/docs/authentication-system.md]: Is this following part covered in the new docs? I couldn't find `addExtension` in https://loopback.io/doc/en/lb4/Loopback-component-authentication.html

Comment (emonddr) [reply packages/authentication/docs/authentication-system.md]: @nabdelgadir 

This is old and was replaced with `registerAuthenticationStrategy`

Comment (emonddr) [reply packages/authentication/docs/authentication-system.md]: https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/types.ts#L65

```ts
export function registerAuthenticationStrategy(
  context: Context,
  strategyClass: Constructor<AuthenticationStrategy>,
) {
  addExtension(
    context,
    AuthenticationBindings.AUTHENTICATION_STRATEGY_EXTENSION_POINT_NAME,
    strategyClass,
    {
      namespace:
        AuthenticationBindings.AUTHENTICATION_STRATEGY_EXTENSION_POINT_NAME,
    },
  );
}

```

Comment (emonddr) [reply packages/authentication/docs/authentication-system.md]: https://loopback.io/doc/en/lb4/Loopback-component-authentication.html#registering-a-custom-authentication-strategy

Comment (nabdelgadir) [reply packages/authentication/docs/authentication-system.md]: Thanks!

**Comment by bajtos** [comment]
> The authentication code example in
packages/authentication/docs/authentication-system.md can be
removed now that
https://loopback.io/doc/en/lb4/Loopback-component-authentication.html

Instead of removing the doc section completely, perhaps we should replace it with a link to `Loopback-component-authentication`?

**Comment by emonddr** [comment]
> > The authentication code example in
> > packages/authentication/docs/authentication-system.md can be
> > removed now that
> > https://loopback.io/doc/en/lb4/Loopback-component-authentication.html
> 
> Instead of removing the doc section completely, perhaps we should replace it with a link to `Loopback-component-authentication`?

Hi @bajtos , @jannyHou  has a separate PR that will clean up the files in the `docs` directory in `packages\authentication` folder.

image

The authentication-system.md in the packages/authentication/docs directory is a `side` document.

The [packages/authentication/README.MD](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/README.md) is the most important `MD` file in for authentication, and it points to authentication-system.md (design decisions) and the official documentation https://loopback.io/doc/en/lb4/Loopback-component-authentication.html. So there is no need for authentication-system.md to point to https://loopback.io/doc/en/lb4/Loopback-component-authentication.html as well.

**Pull request merged**